src/main/java/net/minestom/server/listener/preplay/LoginListener.java

```diff
@@ -163,47 +163,57 @@ private static SecretKey getSecretKey(byte[] sharedSecret) {
         // Proxy support
-        if (connection instanceof PlayerSocketConnection socketConnection) {
-            final String channel = socketConnection.getPluginRequestChannel(packet.messageId());
-            if (channel != null) {
-                boolean success = false;
-
-                SocketAddress socketAddress = null;
-                GameProfile gameProfile = null;
-
-                // Velocity
-                if (VelocityProxy.isEnabled() && channel.equals(VelocityProxy.PLAYER_INFO_CHANNEL)) {
-                    byte[] data = packet.data();
-                    if (data != null && data.length > 0) {
-                        NetworkBuffer buffer = new NetworkBuffer(ByteBuffer.wrap(data));
-                        success = VelocityProxy.checkIntegrity(buffer);
-                        if (success) {
-                            // Get the real connection address
-                            final InetAddress address;
-                            try {
-                                address = InetAddress.getByName(buffer.read(STRING));
-                            } catch (UnknownHostException e) {
-                                MinecraftServer.getExceptionManager().handleException(e);
-                                return;
-                            }
-                            final int port = ((java.net.InetSocketAddress) connection.getRemoteAddress()).getPort();
-                            socketAddress = new InetSocketAddress(address, port);
-                            gameProfile = new GameProfile(buffer);
-                        }
-                    }
-                }
+        if (!(connection instanceof PlayerSocketConnection socketConnection)) {
+            MinecraftServer.LOGGER.warn("Received Login Plugin Respones for non-socket connection " + connection);
+            return;
+        }
 
-                if (success) {
-                    socketConnection.setRemoteAddress(socketAddress);
-                    socketConnection.UNSAFE_setProfile(gameProfile);
-                    CONNECTION_MANAGER.createPlayer(connection, gameProfile.uuid(), gameProfile.name());
-                } else {
-                    LoginDisconnectPacket disconnectPacket = new LoginDisconnectPacket(INVALID_PROXY_RESPONSE);
-                    socketConnection.sendPacket(disconnectPacket);
-                }
-            }
+        String channel = socketConnection.getPluginRequestChannel(packet.messageId());
+
+        LoginPluginMessageBox messageBox = socketConnection.getLoginPluginMessageBox();
+        try {
+            messageBox.handle(packet.messageId(), channel, packet.data());
+        } catch (Throwable t) {
+            MinecraftServer.LOGGER.error("Error handling Login Plugin Response", t);
+            LoginDisconnectPacket disconnectPacket = new LoginDisconnectPacket(ERROR_DURING_LOGIN);
+            socketConnection.sendPacket(disconnectPacket);
+            socketConnection.disconnect();
```

Comment:
The original code doesn't have this disconnect, but I feel just sending the disconnect packet and relying on the client isn't enough?